support additional attributes in `gitlab_project_hook`

RomanNess opened the pull request:
Adds support for `push_events_branch_filter`, `confidential_issues_events` and `confidential_note_events` in `gitlab_project_hook` resource. See related issue #520.

Note that `deployment_events` is not yet supported in the currently vendored version of `go-gitlab`. Once #523 is merged I can prepare a separate PR for `deployment_events`.

armsnyder reviewed (approved):
Thanks for the contribution! BTW #523 was merged.

RomanNess commented:
@armsnyder I've rebased the branch and added support for `deployment_events` as well. 

I'm looking forward to a new provider release that includes these changes, because we have a few projects that depend on the feature to enable full automation with Terraform.
